### PR TITLE
#79 update data service project structure

### LINKED ISSUE #79: Refactor dgraph project structure
Discussion issue for reorganising/refactoring the dgraph project.

## Expected Behavior

For discussion.

## Current Behavior

```
/dgraph
  /graphql
    /lib
    /src
  /sample
  foo.sh
  bar.sh
  ...
```

## Possible Solution

Current thoughts are something like:

```
/data-service
  /database
    /data
    /scripts
      foo.sh
      bar.sh
      ...
  /server
    /lib
    /src
```

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #79.

## Description

Reorganise `dgraph` project and rename to `data-service`.

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [ ] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [x] Tests OK.

### CODE REVIEW

**Comment by wlthomson** [comment]
@katherine-sussol @mark-prins Not sure what happened there... fixed up with a rebase and somehow nothing exploded! 

Looks like some tests still failing so will investigate, probably just need to point them to the new directory.

File: data-service/server/src/resolvers.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+const queries = {
```
Comment (katherine-sussol) [inline data-service/server/src/resolvers.js:1]: Queries look good - I think I'm seeing double though 👓  are these the same as the ones in queries.js?

File: data-service/server/lib/api.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"use strict";
```
Comment (katherine-sussol) [inline data-service/server/lib/api.js:1]: I saw the lib/ directory was already updated in `.gitignore`, is it also worth removing the files from source control?

File: data-service/package.json [hunk 3]
```diff
@@ -16,10 +16,8 @@
     "@babel/preset-env": "^7.10.4"
   },
   "scripts": {
-    "build": "babel graphql/src -d graphql/lib",
-    "start": "node graphql/lib/server.js",
+    "build": "babel server/src -d server/lib",
+    "start": "node server/lib/server.js",
     "test": "echo \"Error: no test specified\" && exit 1"
-  },
-  "author": "mSupply Foundation",
```
Comment (katherine-sussol) [inline data-service/package.json:23]: Hm (this is just a question), is this something that has a convention yet, or pending discussion on a convention? 
I took a quick look through the other projects and noticed quite a bit of a variety. E.g. mobile uses
 "Sustainable Solutions (NZ) Ltd", and V1 of UC has Chris listed as the author 😆 .

Comment (mark-prins) [reply data-service/package.json:23]: No convention yet ( clearly! ) 
The direction I would expect is for everything to be under `The mSupply Foundation` or the `mSupply Foundation`? So yes, still a little discussion required

**Comment by wlthomson** [comment]
@katherine-sussol 

Re. `.gitignore`: `lib` files are removed in https://github.com/openmsupply/unified-codes/pull/88/commits/b999f4164c10e302840a6dc4d52fe109318d5f94 (not included in this PR).

Re. `package.json`: just updated this for consistency with other projects. Since we haven't agreed on how to do this, I think best to keep stuff consistent (can update all the projects at a later time).

**Comment by katherine-sussol** [comment]
Re. `.gitignore` cool - hadn't seen those commits yet 👍
Re. `package.json` fair enough, my question was more out of self-interest next time when creating a new project (and for the nx repo update - will follow suit)
Re. `queries` all good, I'm personally not fussed over same vs separate file at this point (with just 2 queries). Would it make sense to delete the standalone queries.js in this case then, to avoid confusion?

**Comment by wlthomson** [comment]
@katherine-sussol  `queries` file should be deleted (maybe in a later PR, sorry!). Might be best to revisit after all PRs are merged.

**Pull request merged**